### ISSUE TITLE
Null pointer exception may occur here

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#738]
```php
<?php

class Example
{
    public function method()
    {
        printf(null);
        // Warning: Null pointer exception may occur here (result can be null).
        // Actually, any function will make it happen.
    }
}

function test()
{
    strtoupper(null);
    // Works fine here.
}
```

Using _U1.0.3.1_.

**Comment by kalessil** [comment]
I probably need restrain the rule to method calls only, what do you think @rentalhost @funivan ?

**Comment by rentalhost** [comment]
The problem was not that. The problem is any `null` argument is marked with "Null pointer exception may occur here".

**Comment by kalessil** [comment]
Correct, I thought that we shouldn't report cases with core functions at all =).

**Comment by rentalhost** [comment]
I think that is valid report for everything, inclusive inside of functions (currently it check on methods).

**Comment by dkushnikov** [comment]
Found another case:
https://yadi.sk/i/BLfVM0rJ3RaLZf
`
<?php

/** @noinspection AutoloadingIssuesInspection */
class Account {
    public function getInfo(): AccountInfo {
        $info = AccountInfo::get();
        if ($info === null) {
            throw new RuntimeException("Runtime");
        }

        return $info;
    }

    public function getData(): array {
        return [
            'info' => $this->getInfo()->getValue(),
        ];
    }
}

class AccountInfo {
    public static function get(): ?AccountInfo {
        if (true) {
            return new AccountInfo();
        }
        return null;
    }

    public function getValue(): string {
        return '';
    }
}
`

**Comment by kalessil** [comment]
@dkushnikov : which plugin/version are you using?

**Comment by dkushnikov** [comment]
@kalessil 2.3.15.1 / PhpStorm 2017.3.3 EAP Build #PS-173.4301.7

**Comment by kalessil** [comment]
@dkushnikov : git it, most probably it's EAP specific. If I get some time, I'll check the latest EAP, but this will be not high-prio because of pre-release phase.

**Comment by kalessil** [comment]
@rentalhost : fixed your case
@dkushnikov : I created a follow up ticket for you cases - #760
